msal/build.gradle

```diff
@@ -96,6 +107,52 @@ android {
     }
 }
 
+ext.createCoverageTask = { buildVariant, testType ->
+    def unitTest = "test${buildVariant.capitalize()}UnitTest"
+    def androidTest = "connected${buildVariant.capitalize()}AndroidTest"
+    def unitTestFile = "jacoco/${unitTest}.exec"
+    def androidTestFile = "outputs/code_coverage/${buildVariant}AndroidTest/connected/*.ec"
+    def taskList, executionFiles
+    if (testType.matches("BothTest")) {
+        taskList = [unitTest, androidTest]
+        executionFiles = [unitTestFile, androidTestFile]
+    } else if (testType.matches("UnitTest")){
+        taskList = [unitTest]
+        executionFiles = [unitTestFile]
+    } else {
+        taskList = [androidTest]
+        executionFiles = [androidTestFile]
+    }
+
+    task "${buildVariant}Msal${testType}CoverageReport" (type:JacocoReport, dependsOn: taskList) {
```

Comment:
Let's add a little documentation for this task

Reply:
added